### PR TITLE
Close Admin Clients in KafkaSpec Cleanup / Junit testkit improvements

### PR DESCRIPTION
**Pull request by johnclara**
Admin Clients are currently not closed after the KafkaSpec cleanup.
In our tests I see a bunch of warnings:
```
23:35:41.206 [kafka-admin-client-thread | adminclient-1] WARN  o.apache.kafka.clients.NetworkClient - [AdminClient clientId=adminclient-1] Connection to node 0 could not be established. Broker may not be available.
23:35:42.344 [kafka-admin-client-thread | adminclient-1] WARN  o.apache.kafka.clients.NetworkClient - [AdminClient clientId=adminclient-1] Connection to node 0 could not be established. Broker may not be available.
```
Changes in this PR include:
Only one new and old admin client will be used per KafkaSpec.
A new and old admin client will be created even if they're not needed.
The new and old admin clients will be closed on cleanup.

### CODE REVIEW

File: testkit/src/main/scala/akka/kafka/testkit/scaladsl/KafkaSpec.scala [hunk 1]
```diff
@@ -55,6 +56,10 @@ abstract class KafkaSpec(val kafkaPort: Int, val zooKeeperPort: Int, actorSystem
     extends TestKit(actorSystem)
     with KafkaTestKit {
 
+  // children should setup() and use admin or oldAdmin instead of constructors
+  private def adminClient(): AdminClient = super.adminClient()
```
Comment (johnclara) [inline testkit/src/main/scala/akka/kafka/testkit/scaladsl/KafkaSpec.scala]: @2m Is there a better way of removing visibility to children in Scala?

Comment (ennru) [reply testkit/src/main/scala/akka/kafka/testkit/scaladsl/KafkaSpec.scala]: Renaming the factory methods in `internal.KafkaTestKit` to `createXyz` is the best way, I guess as it indicates the caller has some responsibility.

**Comment by johnclara** [comment]
LGTM! Sorry, haven't had a change to look at this recently

**Comment by ennru** [comment]
Thank you @johnclara for initiating quite some improvements to the testkit!

**Pull request merged**